06_puppy/hsy3418/store_test.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+var (
+	toBeCreatedPuppy = func() Puppy {
+		return Puppy{ID: 101, Breed: "Poodle", Colour: "White", Value: 1000.5}
+	}
+	existsPuppy = func() Puppy {
+		return Puppy{ID: 102, Breed: "Poodle", Colour: "White", Value: 1280.5}
+	}
+)
+
+func (suite *storerTestSuite) SetupTest() {
+	suite.store = suite.makeStore()
+	suite.puppy = existsPuppy()
+	err := suite.store.CreatePuppy(suite.puppy)
+	if err != nil {
+		suite.FailNow("Failed to setup test")
+	}
+}
+
+type storerTestSuite struct {
+	suite.Suite
+	store     Storer
+	makeStore func() Storer
+	puppy     Puppy
+}
+
+func (suite *storerTestSuite) TestCreatePuppy() {
+	assert := assert.New(suite.T())
+	testCases := []struct {
+		title    string
+		input    Puppy
+		expected string
+	}{
+		{"Create new puppy", toBeCreatedPuppy(), "<nil>"},
+		{"Create existing puppy", toBeCreatedPuppy(), "puppy with 101 ID already exists"},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		suite.T().Run(tc.title, func(t *testing.T) {
+			err := suite.store.CreatePuppy(tc.input)
+			assert.Equal(tc.expected, fmt.Sprint(err))
```

Comment:
It would be better if you make `expected` type as `error` and compare them.

```go
func (suite *storerTestSuite) TestCreatePuppy() {
	assert := assert.New(suite.T())
	testCases := []struct {
		title    string
		input    Puppy
		expected error
	}{
		{"Create new puppy", toBeCreatedPuppy(), nil},
		{"Create existing puppy", toBeCreatedPuppy(), fmt.Errorf("puppy with 101 ID already exists")},
	}
	for _, tc := range testCases {
		tc := tc
		suite.T().Run(tc.title, func(t *testing.T) {
			err := suite.store.CreatePuppy(tc.input)
			assert.Equal(tc.expected, err)
		})
	}

}

```